src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -52,14 +52,18 @@ export interface AnalysisFile {
 
 export interface AnalysisFileStore {
   refresh: () => Promise<void>
-  loadedState: LoadingState<AnalysisFile[]> | ReadyState<AnalysisFile[]>
+  loadedState: LoadedState<AnalysisFile[], unknown>
+  create: (fullAnalysisName: String, toolLabel: ToolLabel, contents: String) => Promise<void>
+  pendingCreate: Boolean
 }
 
 export const useAnalysisFiles = (): AnalysisFileStore => {
   const signal = useCancellation()
   const [loading, setLoading] = useState(false)
   const workspace: WorkspaceWrapper = useStore(workspaceStore)
   const [analyses, setAnalyses] = useState<AnalysisFile[]>([])
+  const [pendingCreate, setPendingCreate] = useState(false)
```

Comment:
if pendingState is typed as LoadedState<true, unknown>, it can express error cases instead of just true/false state.

Reply:
I played around with it a little bit but couldn't figure out what syntax to use. Could we schedule a meeting?